streaming/index.js

```diff
@@ -121,25 +164,39 @@ const pgConfigFromEnv = (env) => {
         break;
       }
     }
+  } else {
+    throw new Error('Unable to resolve postgresql database configuration.');
   }
 
   return {
     ...baseConfig,
-    max: env.DB_POOL || 10,
+    max: typeof env.DB_POOL === 'string' ? parseInt(env.DB_POOL, 10) : 10,
```

Comment:
We probably need to handle the potential for a `NaN` case here, so lifting it up something like:

```
let db_pool_max = 10;
if (typeof env.DB_POOL === 'string') {
  const parsedValue = parseInt(env.DB_POOL, 10)
  if (isNaN(parsedValue)) {
    throw new Error(`Invalid DB_POOL environment variable, expected integer, received: ${value}`)
  }
  db_pool_max = parsedValue;
}
```